### DISCUSSION TITLE
Is there a wol function?

### DISCUSSION THREAD

**Discussion by dysysqf** [rustdesk/rustdesk#1249] [Q&A] [answered]
Is there a wol function?

**Comment by sanakess** [comment]
wake on lan function would be great. It can works like in anydesk. If one pc in local net is working, it can send magic packet to any pc in this net.

**Comment by rustdesk** [comment, marked as answer]
Same local network required
![image](https://github.com/user-attachments/assets/8891cec4-d391-436b-8de5-968b8893bc88)

There are also other internet solutions.


- pSnap https://github.com/seriousm4x/UpSnap 
- WakeMeOnLAN https://www.nirsoft.net/utils/wake_on_lan.html 
- PingTool https://ping-tool.com/
- Some routers support WOL, https://www.reddit.com/r/rustdesk/comments/1ox1oxv/comment/noun2rj/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

**Reply by nik231180** [reply]
Wake-on-LAN works in a nightly build. But only with a RustDesk ID on macOS on a local network. With a static IP provided by the ISP, with a local IP provided by the router, it doesn't work with any operating system.

**Reply by rustdesk** [reply]
More infomatino to make WOL work.

## Prerequisite
The computer’s network interface card (NIC) and BIOS/UEFI firmware must support Wake-on-LAN. This option also needs to be enabled in both BIOS/UEFI settings and the operating system’s network adapter configuration.

## Discovery
The system performs a LAN scan to detect active devices and record their MAC addresses. This information is used later for WOL operations.

## Execution
To wake up a specific machine, the system sends a special network message, called a magic packet, that includes the target device’s MAC address. When a compatible NIC receives the magic packet, it powers on the system.

## Verification
To check if your system supports WOL, open the network adapter properties in the operating system (for example, Device Manager in Windows → Network Adapter → Properties → Power Management). The adapter should list “Wake on magic packet” or similar capabilities. You can also verify support in the BIOS/UEFI setup under Power Management settings.

https://en.wikipedia.org/wiki/Wake-on-LAN

<img width="597" height="705" alt="image" src="https://github.com/user-attachments/assets/1e23d6d9-14d8-4392-9282-8b2d9087493e" />

Additionally, we use this crate https://github.com/fengyc/wol-rs

You can test if WOL works with these prebuilt binaries. https://github.com/fengyc/wol-rs/releases/tag/v1.1.0

**Comment by deep-soft** [comment]
WOL function works only for computers in local network, same as in AnyDesk does.
It's included also in non nightly version of rusdesk but don't know if it's included in non flutter version.

**Comment by deep-soft** [comment]
Flutter version:
![image](https://github.com/rustdesk/rustdesk/assets/10506197/b561fbf9-a9f3-462a-81c2-79b54ede2cf3)


Sciter version:
![image](https://github.com/rustdesk/rustdesk/assets/10506197/25b39498-05d4-4831-b5b2-0ba4d48bdc29)

**Reply by paulsommer** [reply]
Thanks for your hint. My Linux version (.deb package) also has this menu item (I never saw it consciously), but unfortunately it doesn't work.

**Reply by deep-soft** [reply]
Sciter version has this option for all remotes but Flutter version only for remotes in the Discovered (as I show you in the screenshot above).
The WOL function works only for remotes in the same local network as the Client (computer from which you want to connect).
If the function WOL does not work for local network computers, you must check for these remote computers if the WOL is active.
You can test if WOL is working:
wakeonlan ??:??:??:??:??:??
replace ??:??:??:??:??:?? with an valid MAC address of remote computer
If you don't have wakeonlan program on your Linux computer you can see here:
https://stackoverflow.com/questions/31588035/bash-one-line-command-to-send-wake-on-lan-magic-packet-without-specific-tool

**Comment by sbaum2s** [comment]
I do have this entry in the UI but it doesnt work. The remote computer does not wake up. For some reason the same functionality stopped working in Anydesk as well some time ago. Probably it has to do with a windows update but as far as i can see no settings changed so it remains a mystery. For the other tools mentioned here you need to kow either the mac adress or your ip adress needs to be the same all the time. Usually i do not have the mac adress of the remote computer and my internet service provider may change my ip adress so those solutions are not useful.

**Comment by jannikk09** [comment]
Am I correct that WOL only works if I am on the same network as the computer?
So it doesn't work if another computer (RustDesk is installed there) is turned on in the network?

**Reply by sbaum2s** [reply]
> Am I correct that WOL only works if I am on the same network as the computer? So it doesn't work if another computer (RustDesk is installed there) is turned on in the network?

This scenario works with AnyDesk (in the paid version only). So i can WOL another computer from another network. (AnyDesk client is installed on both computers but that should not be the trick)

**Reply by jannikk09** [reply]
I know that. My goal is to get rid of AnyDesk. I've already purchased a license for RustDesk.

**Comment by rpce** [comment]
If Rustdesk allowed the WOL option to define any port for the magic packet (e.g., instead of the default 9, choose a different port, like 32555), and the ability to enter a MAC address, the router would wake the computer up. If possible, you should assign a static IP address on the router and define a static MAC address assigned to the IP address. This works great on MikroTiks.
